freqtrade/persistence/models.py

```diff
@@ -362,11 +362,54 @@ def _set_stop_loss(self, stop_loss: float, percent: float):
             self.stop_loss_pct = -1 * abs(percent)
         self.stoploss_last_update = datetime.utcnow()
 
-    def set_isolated_liq(self, isolated_liq: float):
+    def set_isolated_liq(
+        self,
+        isolated_liq: Optional[float] = None,
+        wallet_balance: Optional[float] = None,
+        current_price: Optional[float] = None,
+        maintenance_amt: Optional[float] = None,
+        mm_ratio: Optional[float] = None,
+    ):
         """
         Method you should use to set self.liquidation price.
         Assures stop_loss is not passed the liquidation price
         """
+        if not isolated_liq:
+            if not wallet_balance or not current_price:
+                raise OperationalException(
+                    "wallet balance must be passed to LocalTrade.set_isolated_liq when param"
+                    "isolated_liq is None"
+                )
+            if (
+                mm_ratio is None or
+                wallet_balance is None or
+                current_price is None or
+                maintenance_amt is None
+            ):
+                raise OperationalException(
+                    'mm_ratio, wallet_balance, current_price and maintenance_amt '
+                    'required in set_isolated_liq when isolated_liq is None'
+                )
+            isolated_liq = liquidation_price(
```

Comment:
I'd move the calculation of this out of here (you're already doing this same thing in `freqtradebot.py`), so this code will effectively never run.

That will allow the liquidation_price calculations to be moved to the exchange, which will simplify the code quite some (not so many parameters will be necessary, and we can avoid the "if exchange == XXX".